src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1163 @@
+                var osc_calibrate_val = ReadUInt16((byte)Registers.OSC_CALIBRATE_VAL);
+                if (osc_calibrate_val != 0)
+                    periodMilliseconds *= osc_calibrate_val;
+                WriteUInt32((byte)Registers.SYSTEM_INTERMEASUREMENT_PERIOD, (UInt32)periodMilliseconds);
+                WriteRegister((byte)Registers.SYSRANGE_START, 0x04);
+            }
+            else
+            {
+                // If we don't just start the continuous measurement
+                WriteRegister((byte)Registers.SYSRANGE_START, 0x02);
+            }
+        }
+
+        /// <summary>
+        /// Reads the measurement when the mode is set to continuious. 
+        /// </summary>
+        /// <returns>The range in millimeters, a maximum value is returned depending on the various settings</returns>
+        private UInt16 ReadContinuousMeasrurementMillimeters()
+        {
+            Stopwatch stopWatch = Stopwatch.StartNew();
+            var expirationMilliseconds = stopWatch.ElapsedMilliseconds + _operationTimeout;
+            while ((ReadByte((byte)Registers.RESULT_INTERRUPT_STATUS) & 0x07) == 0)
+            {
+                if (stopWatch.ElapsedMilliseconds > expirationMilliseconds)
+                    throw new IOException($"{nameof(ReadContinuousMeasrurementMillimeters)} timeout error");
+            }
+            // assumptions: Linearity Corrective Gain is 1000 (default)
+            // fractional ranging is not enabled
+            var range = ReadUInt16((byte)Registers.RESULT_RANGE_STATUS + 10);
+            WriteRegister((byte)Registers.SYSTEM_INTERRUPT_CLEAR, 0x01);
+            return range;
+        }
+
+        /// <summary>
+        /// Get the distance depending on the measurement mode
+        /// </summary>
+        public UInt16 Distance => MeasurementMode == MeasurementMode.Continuous ? DistanceContinousMillimeters : GetDistanceSingleMillimeters(true);
+
+        /// <summary>
+        /// Get/Set the measurement mode used to return the distance property
+        /// </summary>
+        public MeasurementMode MeasurementMode { get; set; }
+
+        /// <summary>
+        /// Get a distance in millimeters from the continous measurement feature.
+        /// It is recommended to used this method to gethigher quality measurements
+        /// </summary>
+        public UInt16 DistanceContinousMillimeters
+        {
+            get
+            {
+                if (!_continuousInitialized)
+                {
+                    StartContinuousMeasurement();
+                }
+                return ReadContinuousMeasrurementMillimeters();
+            }
+        }
+
+        /// <summary>
+        /// Get a distance in millimeters
+        /// </summary>
+        /// <param name="multipleReads">True if you want multiple try to get a clean value</param>
+        /// <returns>Returns the maximum range so 8190</returns>
+        public UInt16 GetDistanceSingleMillimeters(bool multipleReads = false)
```

Comment:
The name feels a bit weird - perhaps something along the lines of `GetDistanceOnce` and document the unit (we will very likely we will add a Distance struct in the future so this will avoid more changes)

Reply:
Yes, I can change the name. There are 2 modes, continous and single reading. So I kept close to them. But I can clearly rename it like you propose. 
OK to remove the unit and document it then.